dts/bindings/sensor/invensense,icm42605.yaml

```
@@ -0,0 +1,113 @@
+# Copyright (c) 2020 TDK Invensense
+# SPDX-License-Identifier: Apache-2.0
+
+description: ICM-42605 motion tracking device
+
+# ICM-42605 is SPI.
+compatible: "invensense,icm42605"
+
+include: spi-device.yaml
+
+properties:
+    spi-max-frequency:
+      type: int
+      required: true
+      description: Maximum clock frequency of device's SPI interface in Hz
+
+    int-gpios:
+      type: phandle-array
+      required: true
+      description: |
+        The INT signal default configuration is active-high.  The
+        property value should ensure the flags properly describe the
+        signal that is presented to the driver.
+
+    cpha:
+      type: boolean
+      required: true
+      description: |
+        Set to indicate phase starts with asserted half-phase (CPHA=1).
+        For this driver using this property requires also using cpol.
+
+    cpol:
+      type: boolean
+      required: true
+      description: |
+        Set to indicate clock leading edge is falling (CPOL=1).
+        For this driver using this property requires also using cpha.
+
+    accel-hz:
+      type: int
+      required: false
+      default: 12
+      description: |
+        Default frequency of accelerometer.
+      enum:
+        - 12
+        - 25
+        - 50
+        - 100
+        - 200
+        - 500
+        - 1000
+        - 2000
+        - 4000
+        - 8000
+
+    gyro-hz:
+      type: int
+      required: false
+      default: 12
+      description: |
+        Default frequency of gyroscope.
+      enum:
+        - 12
+        - 25
+        - 50
+        - 100
+        - 200
+        - 500
+        - 1000
+        - 2000
+        - 4000
+        - 8000
+
+    accel-fs:
+      type: int
+      required: false
+      default: 0
+      description: |
+        Default full scale of accelerometer.
+        0 - +-16g
+        1 - +-8g
+        2 - +-4g
+        3 - +-2g
+      enum:
+        - 0
+        - 1
+        - 2
+        - 3
+
+    gyro-fs:
+      type: int
+      required: false
+      default: 0
+      description: |
+        Default full scale of gyroscope.
+        0 - +-2000DPS
+        1 - +-1000DPS
+        2 - +-500DPS
+        3 - +-250DPS
+        4 - +-125DPS
+        5 - +-62.5DPS
+        6 - +-31.25DPS
+        7 - +-15.625DPS
+      enum:
+        - 0
```

Note that you can also use a string based enum, e.g.:

```suggestion
        - "2000"
```

Then use the `DT_ENUM_IDX(DT_DRV_INST(i), gyro_fs)` macro to retrieve the enum number. https://github.com/zephyrproject-rtos/zephyr/blob/master/include/devicetree.h#L499

---

@gmarull Thank you for this nice suggestion. 
Implemented and tested. works correctly for fullscale of accel and gyro